examples/device-a.py

```diff
@@ -0,0 +1,13 @@
+from robocluster import Device
+
+device = Device('device-a', '224.0.0.64:32464')
```

Comment:
It would be nice to hide some of the address/port stuff, or at least provide documentation as to what one should use for that parameter.

Reply:
Maybe we could use some kind of hashing algorithm to generate a number between 1-255 and one for 1024-65535?

Reply:
Something like this:
```py
from hashlib import sha256

def to_multicast_group(key):
    """Convert a key to a local multicast group."""
    digest = sha256(key.encode()).digest()

    # 1 byte for last octet of multicast ip
    group = int.from_bytes(digest[0:1], byteorder='little')

    # 2 bytes for port
    port = int.from_bytes(digest[1:3], byteorder='little')
    # make sure port is ephemeral (49152-65535)
    port |= 0xc000

    return '224.0.0.{}:{}'.format(group, port)
```

Reply:
There wouldn't be any benefit of selecting the lsb of the ip and the port number independantly from each other would there? I think this is really good actually, at least for now. We could use URIs to imply a heirarchy of devices.

Reply:
I cannot see any current benefit to selecting ip/port independently.

I'm not sure what you mean by URI hierarchy, but all devices on the same 'group' given by a given key would only be able to subscribe to messages within their current group with the way this is currently implemented.

Reply:
So would everything in our case use the "rover" key? And if we wanted to have multiple groups like "base-station", "rover2", etc, they wouldn't be able to talk to each other directly? I'm just thinking hypotheticaly if we had multiple rovers each with their own gps devices and we wanted them to be updated to each other's position. It would be nice to be able to have a base station subscribe to "rover1/gps/postion", "rover2/gps/position", and rovers could subscribe to "baseStation/joystick1/buttonA_down/", "baseStation/setWaypoint".